src/controllers/scripting/legacy/scriptconnection.cpp

```diff
@@ -5,19 +5,32 @@
 
 void ScriptConnection::executeCallback(double value) const {
     Trace executeCallbackTrace("JS %1 callback", key.item);
-    const auto args = QJSValueList{
-            value,
-            key.group,
-            key.item,
-    };
-    QJSValue func = callback; // copy function because QJSValue::call is not const
-    QJSValue result = func.call(args);
-    if (result.isError()) {
-        if (controllerEngine != nullptr) {
-            controllerEngine->showScriptExceptionDialog(result);
+    if (!m_preventRecursiveCalls) {
```

Comment:
Now you can prevent the callback entirely. 
Was this intended? Does it need a public interface?

Reply:
No, the code is not preventing the callback entirely. The execution of the callback is still happening when the conditions are met.
The code checks if m_preventRecursiveCalls is false and m_callbackExecuting is false before executing the callback. If these conditions are met, the callback function is executed. However, if m_callbackExecuting is true, it means that a connection is being triggered inside the callback, and a critical error message is logged.